spec/unit/lib/gateways/rent_charges_gateway_spec.rb

```diff
@@ -0,0 +1,114 @@
+      address_2: 'fake city',
+      comments: 'comment 2',
+      rr_count: 1,
+      property_type: :House,
+      base_data_bed_size: 3,
+      bedroom_weight: 2,
+      mra_archetype: 'archetype 2',
+      jan_1999_asset_values: 50000,
+      year: 2014,
+      removed: :false,
+      formula_rent_this_year: 160,
+      rent_cap_this_year: 120,
+      uprated_actual: 110
+    }
+  ]}
+
+  it 'creates new rows for the new year' do
+    rent_charges_gateway.save(rent_charges_for_given_year)
+    rent_charges = rent_charges_gateway.all(2014)
+
+    expect(rent_charges[0][:uprn]).to eq('123abc')
+    expect(rent_charges[0][:address]).to eq('1 fake street')
+    expect(rent_charges[0][:address_2]).to eq('fake town')
+    expect(rent_charges[0][:comments]).to eq('comment 1')
+    expect(rent_charges[0][:rr_count]).to eq(1)
+    expect(rent_charges[0][:property_type]).to eq('Flat')
+    expect(rent_charges[0][:base_data_bed_size]).to eq(2)
+    expect(rent_charges[0][:bedroom_weight]).to eq(1)
+    expect(rent_charges[0][:mra_archetype]).to eq('archetype 1')
+    expect(rent_charges[0][:jan_1999_asset_values]).to eq(45000)
+    expect(rent_charges[0][:year]).to eq(2014)
+    expect(rent_charges[0][:formula_rent_this_year]).to eq(170)
+    expect(rent_charges[0][:rent_cap_this_year]).to eq(180)
+    expect(rent_charges[0][:uprated_actual]).to eq(130)
+
+    expect(rent_charges[1][:uprn]).to eq('456def')
+    expect(rent_charges[1][:address]).to eq('2 fake street')
+    expect(rent_charges[1][:address_2]).to eq('fake city')
+    expect(rent_charges[1][:comments]).to eq('comment 2')
+    expect(rent_charges[1][:rr_count]).to eq(1)
+    expect(rent_charges[1][:property_type]).to eq('House')
+    expect(rent_charges[1][:base_data_bed_size]).to eq(3)
+    expect(rent_charges[1][:bedroom_weight]).to eq(2)
+    expect(rent_charges[1][:mra_archetype]).to eq('archetype 2')
+    expect(rent_charges[1][:jan_1999_asset_values]).to eq(50000)
+    expect(rent_charges[1][:year]).to eq(2014)
+    expect(rent_charges[1][:formula_rent_this_year]).to eq(160)
+    expect(rent_charges[1][:rent_cap_this_year]).to eq(120)
+    expect(rent_charges[1][:uprated_actual]).to eq(110)
+  end
+
+  let(:rent_charges_for_different_year) { [
+    { 
+      uprn: Faker::Base.regexify(/[0-9]{3}[a-z]{3}/),
+      address: Faker::Base.regexify(/[0-9]{1,2} [a-z]{5,10}/),
+      address_2: Faker::Base.regexify(/[A-Z]{1}[a-z]{5,10}/),
+      comments: Faker::Base.regexify(/[a-z ]{15}/),
+      rr_count: 1,
+      property_type: 'Flat',
+      base_data_bed_size: Faker::Number.between(1, 3),
+      bedroom_weight: Faker::Number.between(0, 1),
+      mra_archetype: Faker::Base.regexify(/[a-z]{5,10}/),
+      jan_1999_asset_values: Faker::Number.between(20000, 100000),
+      year: 2000,
+      removed: :false,
+      formula_rent_this_year: Faker::Number.between(100, 200),
+      rent_cap_this_year: Faker::Number.between(100, 200),
+      uprated_actual: Faker::Number.between(100, 200)
```

Comment:
No randomised data for rr_count, property_type, year or removed?

Should removed be a symbol or a boolean?

Reply:
I was having issues with property_type because it's an enum. I was doing `Faker::Number.between(0, 5)` for the test data but the tests were failing because the value pulled from the db is the actual enumerated value. I couldn't figure out how to test it using random data?

Reply:
and yeah removed should be a boolean (updated). I was struggling to randomise year and retain control over the tests earlier, but think I've sorted it now (also updated).

Reply:
Nice. You could consider storing the enum in a constant like Models::RentCharge::PROPERTY_TYPES, then using #sample to grab a random type.

Reply:
ah yes, thanks!